library/std/src/ffi/os_str.rs

```diff
@@ -1068,8 +1068,31 @@ impl fmt::Debug for OsStr {
 }
 
 impl OsStr {
-    pub(crate) fn display(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
-        fmt::Display::fmt(&self.inner, formatter)
+    #[stable(feature = "os_str_display", since = "1.51.0")]
+    /// Returns an object that implements [`Display`] for safely printing strings that may contain
+    /// non-Unicode data.
+    ///
+    /// See also [`Path::display()`].
+    ///
+    /// [`Display`]: fmt::Display
+    /// [`Path::display()`]: crate::path::Path::display
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::ffi::OsStr;
+    ///
+    /// let os_str = OsStr::new("foo.txt");
+    /// println!("{}", os_str.display());
+    /// ```
+    ///
+    /// ```
+    /// let path = std::path::Path::new("foo.txt");
+    /// let os_str = path.as_os_str();
+    /// println!("{}", os_str.display());
+    /// ```
```

Comment:
Why this example? For displaying a `Path`, `path.display()` should just be used directly, right?

Reply:
Ok, I changed this to assert that `path.display()` and `path.as_os_str().display()` are the same instead.